docs/_includes/pagination.html

```diff
@@ -1,10 +1,32 @@
 {% if page.prev_page or page.next_page %}
   <div class="mt-10 pt-10 border-t flex flex-col sm:flex-row space-between gap-5">
+    {%- assign url_parts = page.url | split: '/' %}
     {% if page.prev_page %}
-      <a href="{{ page.prev_page.url }}" class="border rounded px-4 py-2 text-left">&lsaquo; {{ page.prev_page.title }}</a>
+    {%- if page.prev_page.url == "index" %}
+       {%- assign prevurl = page.url | replace: url_parts[-1], "" -%}
+    {% else %}
+       {%- assign prevurl = page.url | replace: url_parts[-1], page.prev_page.url -%}
     {% endif %}
+    {%- capture pattern %}page.url == "{{prevurl}}"{% endcapture -%}
+       {%- assign thepage = site.pages | where_exp: "page", pattern | first -%}
+       {%- if thepage.name == "index.md" %}
+          <a href="{{ page.prev_page.url }}" class="border rounded px-4 py-2 text-left">&lsaquo; {{ thepage.title }}</a>
+       {%- else -%}
+          <a href="{{ page.prev_page.url }}" class="border rounded px-4 py-2 text-left">&lsaquo; {{ thepage.title }}</a>
+       {%- endif %}
+    {%- endif %}
```

Comment:
nit: fix indent, simplify code, use "prevpage" as a more meaningful variable name

```suggestion
        {%- if page.prev_page.url == "index" %}
            {%- assign prevurl = page.url | replace: url_parts[-1], "" -%}
        {% else %}
            {%- assign prevurl = page.url | replace: url_parts[-1], page.prev_page.url -%}
        {% endif %}
        {%- capture pattern %}page.url == "{{prevurl}}"{% endcapture -%}
        {%- assign prevpage = site.pages | where_exp: "page", pattern | first -%}
        <a href="{{ page.prev_page.url }}" class="border rounded px-4 py-2 text-left">&lsaquo; {{ prevpage.title }}</a>
    {%- endif %}
```